src/net_processing.cpp

```diff
@@ -2005,18 +2005,27 @@ void PeerManagerImpl::SendPings()
 
 void PeerManagerImpl::RelayTransaction(const uint256& txid, const uint256& wtxid)
 {
-    LOCK(m_peer_mutex);
-    for(auto& it : m_peer_map) {
-        Peer& peer = *it.second;
-        auto tx_relay = peer.GetTxRelay();
-        if (!tx_relay) continue;
```

Comment:
Wouldn't it be better to make this a one-line patch?

`m_next_inv_send_time` will remain `0` for at least as long as the `fSuccessfullyConnected` is false. So an alternative fix would be to check `m_next_inv_send_time==0` in this line.

This should behave the same, because `SendMessages` is run after `fSuccessfullyConnected = true` without processing transactions from different peers in-between.

Reply:
Hm I kinda prefer the current patch. `m_next_inv_send_time` is not meant to be used as a "successfully connected check" so i prefer not to use it in that way here.

Reply:
I agree with Marco that it'd be better to use state in the `Peer` object rather than reaching into the net layer. It feels like that'd make it easier to unit test net procesing (using #25515).

Using the special value `m_next_inv_send_time == 0` to mean "we're not queueing up an inv message for this peer so don't save hashes to send" seems ok to me as long as it's commented. Alternatively, it'd be easy enough to add a bool to `TxRelay` to gate the logic for saving hashes.